### PR TITLE
go14: add go 1.4.2

### PR DESCRIPTION
**Pull request by skatsuta**
Added Go 1.4.2.

### CODE REVIEW

**Comment by dunn** [comment]
Let's add a conflict:
```ruby
conflicts_with "go", :because => "both install a `go` executable"
```

It's that or make this formula keg-only.

File: go14.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
+class Go14 < Formula
+  desc "Go programming environment"
+  homepage "https://golang.org"
+  url "https://storage.googleapis.com/golang/go1.4.2.src.tar.gz"
+  mirror "https://fossies.org/linux/misc/go1.4.2.src.tar.gz"
+  version "1.4.2"
+  sha256 "299a6fd8f8adfdce15bc06bde926e7b252ae8e24dd5b16b7d8791ed79e7b5e9b"
+
+  bottle do
```
Comment (DomT4) [inline go14.rb]: Cut this block. We'll add it on pull.

File: go14.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    sha256 "7913ecbc952e22f9d6e5df114b857ed23ad97adcbc88419fa3b6425b856ee38e" => :mountain_lion
+  end
+
+  option "with-cc-all", "Build with cross-compilers and runtime support for all supported platforms"
+  option "with-cc-common", "Build with cross-compilers and runtime support for darwin, linux and windows"
+  option "without-cgo", "Build without cgo"
+  option "without-godoc", "godoc will not be installed for you"
+  option "without-vet", "vet will not be installed for you"
+
+  deprecated_option "cross-compile-all" => "with-cc-all"
```
Comment (DomT4) [inline go14.rb]: No such thing as a deprecated option in a new formula.

File: go14.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  bottle do
+    sha256 "bb8b8e79201d93eb69e77763535b201ea812d426e259f106e18f62ddf80f86dd" => :yosemite
+    sha256 "46fbe85b2c75e45686ee463eeaa975ce1604f04ee611815d7c163f5feee90e03" => :mavericks
+    sha256 "7913ecbc952e22f9d6e5df114b857ed23ad97adcbc88419fa3b6425b856ee38e" => :mountain_lion
+  end
+
+  option "with-cc-all", "Build with cross-compilers and runtime support for all supported platforms"
+  option "with-cc-common", "Build with cross-compilers and runtime support for darwin, linux and windows"
+  option "without-cgo", "Build without cgo"
+  option "without-godoc", "godoc will not be installed for you"
```
Comment (DomT4) [inline go14.rb]: You either need to remove these two options or include the original:

```ruby
    if build.with?("godoc") || build.with?("vet")
      ENV.prepend_path "PATH", bin
      ENV["GOPATH"] = buildpath
      (buildpath/"src/golang.org/x/tools").install resource("gotools")

      if build.with? "godoc"
        cd "src/golang.org/x/tools/cmd/godoc/" do
          system "go", "build"
          (libexec/"bin").install "godoc"
        end
        bin.install_symlink libexec/"bin/godoc"
      end

      if build.with? "vet"
        cd "src/golang.org/x/tools/cmd/vet/" do
          system "go", "build"
          # This is where Go puts vet natively; not in the bin.
          (libexec/"pkg/tool/darwin_amd64/").install "vet"
        end
      end
    end
```

block.

**Comment by skatsuta** [comment]
@dunn Thank you for your review!

- added the `conflict_with` statement
- cut `bottle` block
- removed the `deprecated_option` statements
- added `godoc` and `vet` blocks (build and test)

File: go14.rb (hunk 3 above)
Comment (skatsuta) [inline go14.rb]: @DomT4 
I added the block, and tests at the end of the code.

**Comment by DomT4** [comment]
Oh. Mike hopped over you today with his Go side-by-side configuration changes in https://github.com/Homebrew/homebrew-versions/commit/0761fbc415db69f97f4fffc291954ea4977540f4. Apologies for that. Thank you for wanting to contribute to Homebrew, we appreciate the submission :heart:

**Pull request closed without merging**